vendor/github.com/docker/docker/client/container_create.go

```diff
@@ -43,14 +42,11 @@ func (cli *Client) ContainerCreate(ctx context.Context, config *container.Config
 	}
 
 	serverResp, err := cli.post(ctx, "/containers/create", query, body, nil)
+	defer ensureReaderClosed(serverResp)
 	if err != nil {
-		if serverResp.statusCode == 404 && strings.Contains(err.Error(), "No such image") {
-			return response, objectNotFoundError{object: "image", id: config.Image}
```

Comment:
it looks like we depend on this in https://github.com/kubernetes/kubernetes/blob/a2c200a57ab9e39be25a2599e98df807b9780818/pkg/kubelet/dockershim/libdocker/helpers.go#L165-L173

does that code need to be adjusted?

Reply:
checking this.

Reply:
https://github.com/moby/moby/pull/38689

Reply:
changes made in https://github.com/kubernetes/kubernetes/pull/75843/commits/b1f1760486b613adfe5bfa50d1d407d12b6f3b03 please take a look